packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -272,28 +325,34 @@ pub(crate) async fn register_indexer_assets(
                     {
                         Ok(result) => {
                             let schema = GraphQLSchema::new(
-                                String::from_utf8_lossy(&data).to_string(),
+                                String::from_utf8_lossy(data).to_string(),
                             );
-                            match schema_manager
-                                .write()
-                                .await
-                                .new_schema(
-                                    &namespace,
-                                    &identifier,
-                                    schema,
-                                    // Only WASM can be sent over the web.
-                                    ExecutionSource::Wasm,
-                                    &mut conn,
-                                )
-                                .await
-                            {
-                                Ok(_) => {
-                                    assets.push(result);
-                                }
-                                Err(e) => {
-                                    let _res =
-                                        queries::revert_transaction(&mut conn).await?;
-                                    return Err(e.into());
+                            // If re are replacing an indexer and keeping its
+                            // data, its schema already exists.
+                            if replace_indexer && !remove_data {
+                                assets.push(result);
```

Comment:
I may be understanding this wrong, but if we're not replacing the schema, can we get away with not doing a DB call at all since the schema is already present in the database?

Reply:
@lostman 
- If someone is uploading without `--remove-data` we need to verify that the schema in the DB (that we aren't changing) matches the schema that's being uploaded
  - Else we will have mismatched schemas on our hands

Reply:
> @deekerno Where is the DB call here?

Actually, I'm pretty sure that I originally misread the code in this section, so my original comment is moot.